tools/release.sh

```diff
@@ -93,14 +93,35 @@ prepare_branch() {
     git fetch --all
     git checkout releases
     git reset --hard origin/releases
-    if [[ "$BRANCH_TO_RELEASE" != "releases" ]] ; then
-        echo "Releasing from other branches than 'releases' is not supported right now."
-        exit -1
-    fi
-
     git clean -xfd
     git submodule update --init --recursive
 
+    # Merge the branch to the releases branch and check the CHANGELOG.md
+    if [[ "$BRANCH_TO_RELEASE" != "releases" ]] ; then
+        git merge "origin/$BRANCH_TO_RELEASE"
+
+        while true
+        do
+            read -r -p "Type the command to edit CHANGELOG.md, default(vim):" editor
+            if [ -z "$editor" ] ; then
+                editor="vim"
+            fi
+            "$editor" CHANGELOG.md
+
+            read -r -p "Please merge the unreleased entries in the 'CHANGELOG.md' and then press any key to continue..." _
+            if [ "$(grep -c "YYYY-MM-DD" CHANGELOG.md)" -eq 1 ] ; then
```

Comment:
this is to check if the changelog entries are merged correctly. eg.:
```
2.2.0 (YYYY-MM-DD)

...

2.1.3 (YYYY-MM-DD)
```

The 2.1.3 should be removed and merge to 2.2.0 sections.